internal/generate/clusterserviceversion/clusterserviceversion_updaters.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusterserviceversion
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/api/pkg/validation"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/version"
+
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	"github.com/operator-framework/operator-sdk/internal/generate/collector"
+)
+
+func applyTo(c *collector.Manifests, csv *operatorsv1alpha1.ClusterServiceVersion) error {
+	// Apply manifests to the CSV object.
+	if err := apply(c, csv); err != nil {
+		return fmt.Errorf("error updating ClusterServiceVersion: %v", err)
+	}
+
+	// Finally sort all updated fields.
+	sortUpdates(csv)
+
+	return validateClusterServiceVersion(csv)
+}
+
+// apply applies the manifests in the collection to csv.
+func apply(c *collector.Manifests, csv *operatorsv1alpha1.ClusterServiceVersion) error {
```

Comment:
Most of this file was copied verbatim from [here](https://github.com/operator-framework/operator-sdk/blob/v0.17.0/internal/generate/olm-catalog/csv_updaters.go)